### DISCUSSION TITLE
Why is afterSignUpUrl or redirect not working?

### DISCUSSION THREAD

**Discussion by Aaryansh1** [clerk/javascript#3348] [Support request - DO NOT OPEN HERE] [answered]
I am using **clerk version 4.29.9** on a **next.js project**
And using **env**, or **afterSignUpUrl** or **afterSignInUrl** or **redirect** none of them are working properly. I am using them on Signup 
 and Signin component
Intended usage is to check if the user is invited or a normal person signing up. -> Then I want to redirect them to either **/verify** or **/invite**

**Comment by CBF2006** [comment, marked as answer]
Yeah I don't know exactly what happened to these commands (were used fine in a tutorial 2 months ago that I'm watching) but I found a fix from [Stack Overflow](https://stackoverflow.com/questions/76309485/clerk-redirecting-to-localhost-after-authentication)

Put these two in your .env.local or .env depending on which you used:
```
NEXT_PUBLIC_CLERK_SIGN_IN_FORCE_REDIRECT_URL="/URL HERE"
NEXT_PUBLIC_CLERK_SIGN_UP_FORCE_REDIRECT_URL="/URL HERE"
``` 

Just replace URL HERE with whatever url you'd like
Don't do it in whatever .tsx file you put the login button in.

For sign out, I just did this in the .tsx file directly
```
<UserButton afterSignOutUrl="/" />
```

Hope this helps!

**Reply by Aaryansh1** [reply]
Thanks for replying,
I did found the exact same solution but decided to go forward with a custom redirect made from middleware. It was much faster and reliable after some configurations.

**Reply by maykon-oliveira** [reply]
I'm facing the same issue. The answer you marked as a solution isn't actually a solution — it just suggests a different approach to avoid the problem rather than solving it.

**Reply by jescalan** [reply]
@maykon-oliveira https://github.com/orgs/clerk/discussions/3348#discussioncomment-10238305

**Comment by crivera** [comment]
Hi - I am having the same problem and see in the URL of the clerk instance, the redirect URL is correct. But after signing in with Google or regular Email it just goes to the default landing page

**Reply by Aaryansh1** [reply]
Maybe check your middleware or something. Are you refreshing the User data correctly? Bcz if it is logging in then user data is not getting refreshed or you have put the / in middleware. Try console logging the conditions.

**Reply by crivera** [reply]
I am not sure I understand - what do you mean with `are you refreshing the user data correctly`?

So i was actually able to reproduce the issue in the default project:

you can do `git clone https://github.com/clerkinc/clerk-nextjs-app-quickstart`

and then create a `.env` file with the following params:

```
NEXT_PUBLIC_CLERK_SIGN_IN_FORCE_REDIRECT_URL=/dashboard1
NEXT_PUBLIC_CLERK_SIGN_UP_FORCE_REDIRECT_URL=/dashboard2

NEXT_PUBLIC_CLERK_PUBLISHABLE_KEY=
CLERK_SECRET_KEY=
```

now do `npm i && npm run dev`

when you log in or sign up you should be redirected to `dashboard1` or `dashboard2` (both pages don't exist - so the 404 page should show) but it actually just goes back the `/`

**Comment by jescalan** [comment]
Hey @crivera - just a note that our staff does not monitor or respond to support requests via github discussions. You may have better luck reaching out to our support team [online](https://clerk.com/contact/support) or [via discord](https://discord.com/invite/b5rXHjAg7A)
